### DISCUSSION TITLE
Idea: `neil dep upgrade`

### DISCUSSION THREAD

**Discussion by teodorlu** [babashka/neil#126] [Ideas]
Original Slack thread on `#babashka-neil`: https://clojurians.slack.com/archives/C03KCV7TM6F/p1660304001426249

Proposed API:

`neil dep upgrade` - upgrades all dependencies to the latest version
`neil dep upgrade --dry-run` - shows all available upgrades
`neil dep upgrade --lib hiccup/hiccup` - upgrades only hiccup
`neil dep upgrade --dry-run --lib hiccup/hiccup` - shows available upgrades for hiccup

**Comment by teodorlu** [comment]
We could possibly allow listing several libraries:

    neil dep upgrade :lib LIBRARY_1 :lib LIBRARY_2

But I'm not sure whether that would cause problems for the global `:lib` option spec.

**Comment by borkdude** [comment]
For now let's not add more stuff to the issue, and focus on what's described in the first post.
